specification/network/resource-manager/Microsoft.Network/stable/2018-08-01/examples/VirtualNetworkGetWithSubnetDelegation.json

```diff
@@ -0,0 +1,54 @@
+{
+  "parameters" : {
+    "api-version": "2018-08-01",
+    "subscriptionId" : "subId",
+    "resourceGroupName" : "rg1",
+    "virtualNetworkName" : "test-vnet"
+  },
+  "responses" : {
+    "200" : {
+      "body" : {
+        "name" : "test-vnet",
+        "id" : "/subscriptions/subId/resourceGroups/rg1/providers/Microsoft.Network/virtualNetworks/test-vnet",
+        "type" : "Microsoft.Network/virtualNetworks",
+        "location" : "westus",
+        "properties" : {
+          "provisioningState" : "Succeeded",
+          "addressSpace" : {
+            "addressPrefixes" : [
+              "10.0.0.0/16"
+            ]
+          },
+          "subnets" : [
+            {
+              "name" : "subnet1",
+              "id" : "/subscriptions/subId/resourceGroups/rg1/providers/Microsoft.Network/virtualNetworks/test-vnet/subnets/subnet1",
+              "properties" : {
+                "provisioningState" : "Succeeded",
+                "addressPrefix" : "10.0.1.0/24",
+                "ipConfigurations" : [
+                  {
+                    "id" : "/subscriptions/subId/resourceGroups/rg1/providers/Microsoft.Network/loadBalancers/lb/frontendIPConfigurations/fe"
+                  }
+                ],
+                "delegations": [
+                  {

```

Comment:
According to the spec, "delegations" is a list of `ServiceDelegationPropertiesFormat` which doesn't contain name, id, properties.

Reply:
Hey Jianghao - I noticed that "id" isn't included in most of the resources, but is included in the examples (as it is in the response of the request).

Reply:
Does your service return "id", "name", etc? The spec and examples must match the service behavior.

Reply:
"provisioningState" in "properties" is not defined in the spec.

Reply:
Id, name comes from the SubResource.
 
And provisioningState is included here:
    "ServiceDelegationPropertiesFormat": {
      "properties": {
        "serviceName": {
          "type": "string",
          "description": "The name of the service to whom the subnet should be delegated (e.g. Microsoft.Sql/servers)"
        },
        "actions": {
          "type": "array",
          "items": {
            "type": "string"
          },
          "description": "Describes the actions permitted to the service upon delegation"
        },
        "provisioningState": {
          "type": "string",
          "description": "The provisioning state of the resource."
        }
      },